strider/fetcher.py

```diff
@@ -179,6 +182,7 @@ async def execute_step(
             self,
             step: Step,
             curie: str,
+            category: str = None,
```

Comment:
I think we should make this `category: str` to enforce that the category parameter is at least passed in, even if it is None.